tests/test_transactions.py

```
@@ -0,0 +1,177 @@
+        c1.send(["FLUSHDB"])
+        assert_eq(c1.send(["SET", "auto_key", "auto_value"]), "+OK", f"{c1.name}: SET auto_key outside transaction")
+        assert_eq(c2.get_value("auto_key"), "auto_value", f"{c2.name}: auto_key should be immediately visible (auto-commit)")
+        print("[PASS] DML outside transaction tests complete.")
+
+        # --- 8. Cleanup ---
+        print("\n-- Phase 8: Cleanup --")
+        c1.send(["WIPEDB"])
+        print("[PASS] Cleanup complete.")
+
+    finally:
+        if cursor1: cursor1.close()
```

Ruff (0.13.1)

Multiple statements on one line (colon)

(E701)